### PR TITLE
Fidelis-SSv2

### PR DESCRIPTION
**Pull request by securesynapse**
## Status
Ready

## Related Issues
N/A

## Description
Added ability to pull session data, pull decoding data, add comment to an alert, pull execution forensic data, assign an explicit score, change open/close status, assign alert to a user, pull forensic data, force file scanning for a file that exists in Fidelis, add/remove/change alert label.

Some code exists for pushing GeoIP changes, File uploading and URI scanning. Code that pertains to those three items does not currently work in this version. The code that does work does not affect the operation of the Fidelis/Demisto integration.

## Screenshots
Paste here any images that will help the reviewer

## Required version of Demisto
19.x

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
None

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

File: Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import demistomock as demisto
-from CommonServerPython import *
-from CommonServerUserPython import *
+
+
+
```
Comment (anara123) [inline Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py]: return these lines
```suggestion
import demistomock as demisto	
from CommonServerPython import *	
from CommonServerUserPython import *
```

File: Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py [hunk 2]
```diff
@@ -176,6 +177,341 @@ def generate_time_settings(time_frame=None, start_time=None, end_time=None):
 
 ''' COMMANDS + REQUESTS FUNCTIONS '''
 
+##Code added by Dave for additional functionality - Second set of items
+
+#This section will change the explicit score of an alert - works
```
Comment (anara123) [inline Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py]: All the function descriptions should be in google docstring format, for example:
```python
def update_alertstatus_command():
    """
    This section will change the explicit score of an alert - works
    """
```

File: Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py [hunk 3]
```diff
@@ -176,6 +177,341 @@ def generate_time_settings(time_frame=None, start_time=None, end_time=None):
...
+    data = {
+        'alertIds': [alert_id],
+        'explicitScore': explicitScore
+    }
+
+    result = update_alertstatus(alert_id, data)
+
+@logger
+def update_alertstatus(alert_id, data):
+    #url = '/j/rest/v1/alert/feedback/{}/'.format(alert_id)
```
Comment (anara123) [inline Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py]: remove unnecessary comments

File: Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py [hunk 4]
```diff
@@ -176,6 +177,341 @@ def generate_time_settings(time_frame=None, start_time=None, end_time=None):
...
+    alert_id = args['alert_id']
+
+    result = alert_ef_submission(alert_id)
+
+    output = {
+        'ID': alert_id,
+        'sessiondata': result
+     }
+
+    demisto.results({
```
Comment (anara123) [inline Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py]: replace `demisto.results` with `return_outputs`
```python
return_outputs(
    readable_output=tableToMarkdown('Alert {}'.format(alert_id), result, headerTransform=pascalToSpace, removeNull=True),
    outputs={
        'Fidelis.Alert(val.ID && val.ID == obj.ID)': output,
    },
    raw_response=result
)
```
change in all the places accordingly

Comment (anara123) [reply Integrations/FidelisElevateNetwork/FidelisElevateNetwork-SS.py]: code conventions
https://github.com/demisto/content/tree/master/docs/code_conventions

**Comment by anara123** [comment]
Hello @securesynapse 
Thank you for your contribution. Please make sure you follow our [contribution guide](https://github.com/demisto/content/blob/master/CONTRIBUTING.md)

Currently your build is failing, make sure it passes

**Comment by anara123** [comment]
We have new dev documentation website https://demisto.pan.dev/

**Comment by Arsenikr** [comment]
@securesynapse can you assist with the link to the API documentation? i assume it is somewhere on the instance but i can't find it

**Comment by securesynapse** [comment]
> @securesynapse can you assist with the link to the API documentation? i assume it is somewhere on the instance but i can't find it

Most of the code I wrote is not documented in the API guide. I have been using google developer tools and looking for the API calls and constructing my code that way.

**Comment by roysagi** [comment]
Hi @securesynapse,

I went through the PR and made some changes in the yml and py files in order to fit them to the Demisto's code standard.

I have some issues and questions that were raised when checking:
1) I added the context output to the yml file but I could not find descriptions to the fields. Do you by any chance have these descriptions?

2) Some commands did not work as intended on our instance - mostly because our alerts do not have dpath or exectution forensics info. One command did not work at all though it should have - the manage-alert-label command. Can we please schedule a demo so we can see these commands in action?

@ronykoz, @Arsenikr - FYI

**Comment by ronykoz** [comment]
Merging this into an inner branch, where the build problems will be solved, @securesynapse thanks for the contribution, and @roysagi will take it from here.

**Pull request merged**